robot/lib/Inc/drivers/ISM43340.hpp

```diff
@@ -371,7 +371,8 @@ class ISM43340 : public GenericRadio {
 
     void pingRouter();
 
-    bool isConnected() const { return isInit; }
+    bool isInitialized() const { return isInit; }
+    bool isConnected() { return connected; }
```

Comment:
`isConnected` and maybe `isInitialized` are both generic enough that we can put this into the `GenericRadio` interface. This way `isConnected` in `RadioLink` is not based on receiving a packet, but on connecting to the wifi network

Reply:
I don't quite get the second sentence. How would moving these methods to the interface help? Wouldn't they still have to defined individually in the subclasses?

Reply:
If you move it up to the interface, we can call it from the radio link class and convert that `radioConnected` flag to be representing the `isConnected()` from the driver (in our case that is connected to the network) instead of just representing receiving a packet. Same with `isInitialized()`. It basically allows errors from the drivers to get pushed up into the higher level things, anything not in the interface is locked into the driver and can never be exposed